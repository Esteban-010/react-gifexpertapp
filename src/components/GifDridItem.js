import React from 'react'
import PropTypes from 'prop-types';

export const GifDridItem = ({ title, url }) => {
    

    
    return (
        <div className="card animate__animated animate__fadeIn" >
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </div>
    )
}

GifDridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
