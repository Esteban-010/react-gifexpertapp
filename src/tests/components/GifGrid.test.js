import React from 'react'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en GifGrid.js', () => {
    const category = 'Naruto';

    test('debe motrar correctamente el componente ', () => {
        
        useFetchGifs.mockReturnValue({
            data : [],
            loading : true
        })
        
        const wrapper = shallow(<GifGrid category={ category }/>)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquie/cosa.jpg',
            title: 'Cualquie cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquie/cosa.jpg',
            title: 'Cualquie cosa'
        }];
        
        useFetchGifs.mockReturnValue({
            data : gifs,
            loading : false
        })
        
        const wrapper = shallow(<GifGrid category={ category }/>)
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifDridItem').length).toBe(gifs.length);

    })
    
})
