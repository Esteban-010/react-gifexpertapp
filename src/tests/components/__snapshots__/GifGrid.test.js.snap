// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid.js debe motrar correctamente el componente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
